analyzer/protocol/ldap/asn1.spicy

```diff
@@ -0,0 +1,339 @@
+
+  len: LengthType {
+    self.total_length = self.len.len + self.tag.len + self.len.tag_len;
+  }
+};
+
+#- ASN.1 message body ---------------------------------------------------------
+
+public type ASN1Body = unit(head: ASN1Header, recursive: bool, depth: uint32) {
+  switch ( head.tag.tpe ) {
+
+    ASN1Type::BOOLEAN -> bool_value: uint8 &convert=cast<bool>($$) {
+      if (head.len.len != 1) {
+        throw "ASN1Type::BOOLEAN detected with length != 1";
+      }
+    }
+
+    ASN1Type::INTEGER,
+    ASN1Type::ENUMERATED -> num_value: bytes &size=head.len.len
+                                             &convert=$$.to_int(spicy::ByteOrder::Big);
+
+    ASN1Type::NULLVAL -> : void {
+      if (head.len.len != 0) {
+        throw "ASN1Type::NULLVAL detected with non-zero length";
+      }
+    }
+
+    ASN1Type::BITSTRING -> bitstring: ASN1BitString(head.len.len, head.tag.constructed);
+
+    ASN1Type::OCTETSTRING -> octetstring: ASN1OctetString(head.len.len, head.tag.constructed)
+                                          &convert=$$.value.decode(hilti::Charset::ASCII);
+
+    ASN1Type::OBJECTIDENTIFIER -> objectidentifier: ASN1ObjectIdentifier(head.len.len)
+                                                    &convert=$$.oidstring;
+
+    ASN1Type::PrintableString,
+    ASN1Type::GeneralizedTime,
+    ASN1Type::UTCTime,
+    ASN1Type::UTF8String,
+    ASN1Type::BMPString,
+    ASN1Type::UniversalString -> asn1string: ASN1String(head.tag, head.len.len)
+                                             &convert=$$.value;
+
+    ASN1Type::SEQUENCE, ASN1Type::SET -> seq: ASN1SubMessages(head.len.len, depth+1) if (recursive);
+
+    * -> : bytes &size=head.len.len {
+      throw "Unsupported ASN1Body type";
+    }
+  };
+
+  on %done {
+    if (DEBUG_PRINT == True) {
+      if (self?.bool_value) {
+        print "%.*s%s %s %s" % ((depth*2)+1, DEBUG_SPACES, head.tag.class, head.tag.tpe, "BOOLEAN"), self.bool_value;
+      } else if (self?.num_value) {
+        print "%.*s%s %s %s" % ((depth*2)+1, DEBUG_SPACES, head.tag.class, head.tag.tpe, "NUMBER"), self.num_value;
+      } else if (self?.bitstring) {
+        print "%.*s%s %s %s" % ((depth*2)+1, DEBUG_SPACES, head.tag.class, head.tag.tpe, "BITSTRING"), self.bitstring, head.tag.constructed;
+      } else if (self?.octetstring) {
+        print "%.*s%s %s %s" % ((depth*2)+1, DEBUG_SPACES, head.tag.class, head.tag.tpe, "OCTETSTRING"), self.octetstring, head.tag.constructed;
+      } else if (self?.objectidentifier) {
+        print "%.*s%s %s %s" % ((depth*2)+1, DEBUG_SPACES, head.tag.class, head.tag.tpe, "OBJECTIDENTIFIER"), self.objectidentifier;
+      } else if (self?.asn1string) {
+        print "%.*s%s %s %s" % ((depth*2)+1, DEBUG_SPACES, head.tag.class, head.tag.tpe, "STRING"), self.asn1string;
+      } else if (self?.seq) {
+        print "%.*s%s %s %s" % ((depth*2)+1, DEBUG_SPACES, head.tag.class, head.tag.tpe, "SEQUENCE|SET"), |self.seq.submessages|, head.len.len;
```

Comment:
This combination of stringification, tuples, and`|...|` triggers the CI error you are seeing.

We have the PR zeek/spicy#919 open to fix this; alternatively and as a temporary workaround you could pull the size computation into a variable and print that,i.e.,

```.r
local num_submessages = |self.seq.submessages|;
print ... % ... num_submessages ...;
```

If you decide to not wait, but go with a workaround, leaving a `TODO` referencing the PR would be great.

Reply:
I put the workaround in as you suggested, with a TODO referencing the PR. Thanks. 7737a0c